src/blueprint/from_config.rs

```diff
@@ -223,6 +225,56 @@ fn to_type(name: &str, list: &Option<bool>, required: &Option<bool>, list_type_r
   }
 }
 
+fn validate_query(config: &Config) -> Valid<()> {
+  let query_type_name = config
+    .graphql
+    .schema
+    .query
+    .as_ref()
+    .validate_some("Query root is missing".to_owned())?;
+
+  let Some(query) = config.find_type(query_type_name) else {
+    return Valid::fail("Query type is not defined".to_owned()).trace(query_type_name);
+  };
+
+  query
+    .fields
+    .iter()
+    // validate only required fields
+    // TODO: somehow when field is required it has a value of Option<false> due to https://github.com/tailcallhq/tailcall/blob/49e39e12013cf837ce0cc78562c4207f4a9782ad/src/config/from_document.rs#L195
+    .filter(|(_, field)| field.required.is_some())
```

Comment:
```suggestion
```

On second thoughts we should throw this error irrespective of the nullability.

Reply:
Apologies on going to and fro on this 🙈

Reply:
no problem, I've reverted these changes